NCS.DSS.Customer/Cosmos/Client/DocumentDBClient.cs

```diff
@@ -13,12 +13,44 @@ public static DocumentClient CreateDocumentClient()
             if (_documentClient != null)
                 return _documentClient;
 
-            _documentClient = new DocumentClient(new Uri(
-                ConfigurationManager.AppSettings["Endpoint"]),
-                ConfigurationManager.AppSettings["Key"]);
+            _documentClient = InitialiseDocumentClient();
 
             return _documentClient;
         }
-        
+
+        private static DocumentClient InitialiseDocumentClient()
+        {
+            string connectionString;
+
+            try
+            {
+                connectionString = ConfigurationManager.AppSettings["CustomerConnectionString"];
+
+            }
+            catch (Exception e)
+            {
+                throw;
+            }
+
+            if(string.IsNullOrWhiteSpace(connectionString))
+                throw new ArgumentNullException();
+
+            var endPoint = connectionString.Split(new[] { "AccountEndpoint=" }, StringSplitOptions.None)[1]
```

Comment:
Will this throw a useful error if the connection string is badly formatted and the split doesn't work as expected?  If not it should be wrapped in a try catch

Reply:
missed commit